crates/editor/src/editor.rs

```diff
@@ -1157,6 +1160,10 @@ impl Editor {
                 project_subscriptions.push(cx.subscribe(project, |editor, _, event, cx| {
                     if let project::Event::RefreshInlayHints = event {
                         editor.refresh_inlay_hints(InlayHintRefreshReason::RefreshRequested, cx);
+                    } else if let project::Event::LanguageServerAdded(_, _, _) = event {
+                        editor.tasks_pull_diagnostics_task = Some(editor.refresh_diagnostics(cx));
+                    } else if let project::Event::LanguageServerRemoved(_) = event {
+                        editor.tasks_pull_diagnostics_task = Some(editor.refresh_diagnostics(cx));
```

Comment:
**Question:** Not sure if triggering the task for pulling diagnostics here is the correct approach. My thinking is with the `MultiLspQuery` we want to pull new diagnostics sostics every time new LSP is added or removed.